### ISSUE TITLE
Engine test failing with `no-deprecated --api=1.1.1` in version 3.0.1

### ISSUE THREAD

**Issue by MarcelRaad** [openssl/openssl#17649]
`./config no-deprecated --api=1.1.1 && make` gives me the following compiler warnings with OpenSSL 3.0.1:

```
In file included from test/enginetest.c:18:
test/enginetest.c: In function 'test_x509_dup_w_engine':
test/enginetest.c:367:29: warning: implicit declaration of function 'PEM_read_bio_X509' [-Wimplicit-function-declaration]
  367 |         || !TEST_ptr(cert = PEM_read_bio_X509(b, NULL, NULL, NULL)))
      |                             ^~~~~~~~~~~~~~~~~
test/testutil.h:456:64: note: in definition of macro 'TEST_ptr'
  456 | # define TEST_ptr(a)          test_ptr(__FILE__, __LINE__, #a, a)
      |                                                                ^
test/enginetest.c:367:27: warning: assignment to 'X509 *' {aka 'struct x509_st *'} from 'int' makes pointer from integer with>  367 |         || !TEST_ptr(cert = PEM_read_bio_X509(b, NULL, NULL, NULL)))
      |                           ^
test/testutil.h:456:64: note: in definition of macro 'TEST_ptr'
  456 | # define TEST_ptr(a)          test_ptr(__FILE__, __LINE__, #a, a)
      |                                                                ^
test/enginetest.c:403:27: warning: assignment to 'X509 *' {aka 'struct x509_st *'} from 'int' makes pointer from integer with>  403 |         || !TEST_ptr(cert = PEM_read_bio_X509(b, NULL, NULL, NULL)))
      |                           ^
test/testutil.h:456:64: note: in definition of macro 'TEST_ptr'
  456 | # define TEST_ptr(a)          test_ptr(__FILE__, __LINE__, #a, a)
```

And then `make test` gives me the following test failure:
```
30-test_engine.t ...................
../../util/wrap.pl ../../test/enginetest ../../test/certs/root-cert.pem => 139
not ok 1 - running enginetest
30-test_engine.t ................... 1/? ---------------------------------------
#   Failed test 'running enginetest'
#   at test/recipes/30-test_engine.t line 24.
30-test_engine.t ................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
```

`perl configdata.pm --dump` output from my Alpine Linux 3.13 system (the same also happens with my Debian Linux 11 system):
```
Command line (with current working directory = .):

    perl ./Configure --api=1.1.1 no-deprecated

Perl information:

    perl
    5.32.0 for x86_64-linux-thread-multi

Enabled features:

    afalgeng
    aria
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests          [cascade]        OPENSSL_NO_ACVP_TESTS
    asan                [default]        OPENSSL_NO_ASAN
    buildtest-c++       [default]
    crypto-mdebug       [default]        OPENSSL_NO_CRYPTO_MDEBUG
    deprecated          [option]         OPENSSL_NO_DEPRECATED
    devcryptoeng        [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]        OPENSSL_NO_EGD
    external-tests      [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                [default]
    fips-securitychecks [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl            [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]        OPENSSL_NO_KTLS
    md2                 [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]        OPENSSL_NO_MSAN
    rc5                 [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]        OPENSSL_NO_SCTP
    trace               [default]        OPENSSL_NO_TRACE
    ubsan               [default]        OPENSSL_NO_UBSAN
    unit-test           [default]        OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]
    zlib-dynamic        [default]
    ssl3                [default]        OPENSSL_NO_SSL3
    ssl3-method         [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86_64",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread -m64",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =
    CPP =
    CPPDEFINES =
    CPPFLAGS =
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =
    HASHBANGPERL =
    LD =
    LDFLAGS =
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL =
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =

Makevars:

    AR              = ar
    ARFLAGS         = qc
    CC              = gcc
    CFLAGS          = -Wall -O3
    CPPDEFINES      =
    CPPFLAGS        =
    CPPINCLUDES     =
    CXX             = g++
    CXXFLAGS        = -Wall -O3
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         =
    LDLIBS          =
    PERL            = perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         =

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
```

On both systems, the test succeeds with just `--api=1.1.1` without `no-deprecated` and is skipped with `no-deprecated --api=3.0`.

**Comment by mattcaswell** [comment]
Fix in #17652.

**Comment by MarcelRaad** [comment]
Thanks for the quick fix!
